src/driver.rs

```diff
@@ -466,4 +552,46 @@ mod tests {
         assert!(DriverManager::count() > 0);
         assert!(DriverManager::get_driver(0).is_ok());
     }
+
+    #[test]
+    fn test_driver_extension() {
+        // convert the driver into short_name for testing purposes
+        let drivers = |filename, is_vector| {
+            DriverManager::get_drivers_for_filename(filename, is_vector)
+                .iter()
+                .map(|d| d.short_name())
+                .collect::<HashSet<String>>()
+        };
+        let gdal_version: i64 = crate::version::version_info("VERSION_NUM").parse().unwrap();
+        if DriverManager::get_driver_by_name("ESRI Shapefile").is_ok() {
+            assert!(drivers("test.shp", true).contains("ESRI Shapefile"));
+            // `shp.zip` only supported from gdal version 3.1
+            // https://gdal.org/drivers/vector/shapefile.html#compressed-files
+            if gdal_version >= 3010000 {
+                assert!(drivers("test.shp.zip", true).contains("ESRI Shapefile"));
+            }
+        }
+        if DriverManager::get_driver_by_name("GPKG").is_ok() {
+            assert!(drivers("test.gpkg", true).contains("GPKG"));
+            // `gpkg.zip` only supported from gdal version 3.7
+            // https://gdal.org/drivers/vector/gpkg.html#compressed-files
+            if gdal_version >= 3070000 {
+                assert!(drivers("test.gpkg.zip", true).contains("GPKG"));
+            }
+        }
+        if DriverManager::get_driver_by_name("GTiff").is_ok() {
+            assert!(drivers("test.tiff", false).contains("GTiff"));
+        }
+        if DriverManager::get_driver_by_name("netCDF").is_ok() {
+            assert!(drivers("test.nc", false).contains("netCDF"));
+        }
+        if DriverManager::get_driver_by_name("Elasticsearch").is_ok() {
+            // Elasticsearch short name was ElasticSearch in older versions
+            if gdal_version >= 3010000 {
+                assert!(drivers("ES:test", true).contains("Elasticsearch"));
+            } else {
+                assert!(drivers("ES:test", true).contains("ElasticSearch"));
+            }
+        }
+    }
```

Comment:
Can you add tests for files like `this.is.my.test.tiff` and `my.other.test.shp.zip`?

Reply:
done